pkg/trustedresources/errors.go

```diff
@@ -25,3 +27,40 @@ var (
 	// ErrRegexMatch is returned when regex match returns error
 	ErrRegexMatch = errors.New("regex failed to match")
 )
+
+type verificationResultType int
+
+const (
+	verificationResultError = iota
+	verificationResultWarn
+	verificationResultPass
+)
+
+// VerificationError wraps the condition with the error
+type VerificationError struct {
```

Comment:
Here's what I was picturing (feel free to tweak):

```
var (
   ErrNoMatchWarn = errors.New("warning: no matching policies")
   ErrNoMatchFail = errors.New("verification failed: no matching policies")
   ErrVerificationFailed = errors.New("failure: verification failed; mismatched signature")
   ErrVerificationWarn = errors.New("warning: verification failed; mismatched signature")
)

type VerificationError struct {
    err error
}

func (e *VerificationError) Unwrap() error {return e.err}
func (e *VerificationError) Error() string {return e.err.Error()}
```

Then `trustedResources.VerifyTask` could return a VerificationError wrapping one of the other error types defined here, and calling functions can use e.g. `errors.Is(ErrVerificationWarn)`. Or alternatively:

```
func IsTrustedResourcesFailure(e error) bool { return errors.Is(e, ErrNoMatchFail) || errors.Is(e, ErrVerificationFailed) 

func IsTrustedResourcesWarning(e error) bool { return errors.Is(e, ErrNoMatchWarn) || errors.Is(e, ErrVerificationWarn) 
```

Do you think something like that would work? @wlynch wdyt?

Reply:
Thanks for the detailed suggestion!
This seems duplicate? I think current approach just require 1 error for each type and can meet the same need

Reply:
The thought process here was to be able to better inform the user about what went wrong, so they can fix it. If there were no matching policies when there should have been, they can double check whether the policies match the resources, otherwise they can double check that their resources are signed correctly.

You're right though that this info could just go in the error message, and that we could just have ErrVerificationFailed and ErrVerificationWarn.

Reply:
Generally sgtm (enum is also fine), though we should be looser with the definition of verification failure. Any error from verify should be a verification failure, unless it's explicitly warn. This way if there are other errors that don't end up getting wrapped (network errors, timeouts, context cancellations) we don't accidentally say verification passed because we didn't see a VerificationError.